thirdeye-server/src/main/java/ai/startree/thirdeye/service/AnomalyService.java

```diff
@@ -78,9 +84,18 @@ protected AnomalyDTO createDto(final ThirdEyePrincipal principal,
     return toDto(api);
   }
 
+  @Override
+  protected void validate(final AnomalyApi api, @Nullable final AnomalyDTO existing) {
+    super.validate(api, existing);
+    ensureNull(api.getAuth(), "cannot set auth for anomalies");
+  }
+
   @Override
   protected AnomalyDTO toDto(final AnomalyApi api) {
-    return ApiBeanMapper.toDto(api);
+    final var dto = ApiBeanMapper.toDto(api);
+    final var authId = authorizationManager.resourceId(dto);
+    dto.setAuth(new AuthorizationConfigurationDTO().setNamespace(authId.namespace));
```

Comment:
cosmetic, 

Let's encapsulate to `auth.getNamespace()`

can be done in separate PR.